core/src/toga/app.py

```diff
@@ -236,6 +158,7 @@ def __init__(
         home_page: str | None = None,
         description: str | None = None,
         startup: AppStartupMethod | None = None,
+        document_types: dict[str, type[Document]] | None = None,
```

Comment:
Could we move `App.document_types` to `DocumentSet.types`, so all the document stuff is contained in that one object?

Reply:
[Reply by @freakboy3742 moved from other thread]

I get the inclination to put all the document management stuff in one namespace; but to me, app.documents.types reads more as the types of the specific document instances that have been opened, whereas app.document_types reads more as intended - the types of document this app can manage.

Reply:
Both of them could be interpreted both ways, but it's probably too much to ask that the meaning of every API can be guessed just by looking at its name. As soon as the developer looks at the documentation, I think it'll be obvious enough.

Putting it all in one namespace also means it'll all be on the same page of the documentation, which will make it easier to learn. And it keeps the size of the App class down.

Reply:
That's fair; and, when combined with the changes to extension handling, it makes even more sense.